### DISCUSSION TITLE
Working with Source Code

### DISCUSSION THREAD

**Discussion by najak3d** [dotnet/Comet#214] [Q&A]
Can you put up a "Getting Started" guide for those trying to work with source code.   That's what I'm attempting, but without success.  Both here and for Maui.Graphics.Controls (another sweet Maui project).

Our goal is to combine use of BOTH Comet and Maui.Graphics.Controls.  Make them work together.

So we figured it might be good practice to start working inside your source code project directly, and see the innards at work.

**Comment by dgellow** [comment]
I would also be interested in a README or guide discussing the codebase. I already spent some time digging into the code on my side, and had trouble understanding the project.

A things I wrote down when reading the codebase:
1. There are a lot of different *.sln files, such as `Comet.sln`, `Comet.WinUI.sln`, `Comet-net6.sln`, `Comet.Blazor.sln`, and 5 more in the `build` directory. What is the reason for all those solutions? Are they all relevant? Which one should I use if I want to try to contribute? Some seem to be outdated, or incomplete (for example `Comet.WinUI.sln` looks experimental?).
2. Controls are implemented in `src/Comet/Handlers/*`, with one file per platform, such as `NavigationViewHandler.Android.cs`. But there is also a navigation control implemented in `src/Comet/Platform/Android/CometNavigationView.cs`. What is the relationship between them, and when is it necessary to have something in the `Platform` directory?
3. A lot of source generators seem to be used, but it's not always clear what they do. That could benefit from documentation.

**Reply by saint4eva** [reply]
Indeed, the project structure needs to be reorganised for easier understanding and navigation. I think the most important folder is in src/Comet, and its subfolders Controls and Handlers. But it seems most of the controls are generated by Source Generators.

The readme should also state what is remaining to have feature parity with .NET Maui - especially the controls. The more understandable the project structure is, the easier for us to contribute and make PRs
